### DISCUSSION TITLE
What does a solid red LED on the front right arm of the CrazyFlie mean? Is there a pm log value that corresponds?

### DISCUSSION THREAD

**Discussion by jsolderitsch** [bitcraze/discussions#1507] [Q&A]
Writing some python code to use MotionCommander. I want a way of knowing how fresh the battery that is connected to the CrazyFlie is. pm.batteryLevel looked promising but when I execute a takeoff action, the batteryLevel is logged as 10 even when the battery was freshly charged. If I print the value of batteryLevel before takeoff it is higher: 90 when the battery is completely freshly charged and lower but still above 50 when the battery is less fresh.

Can I leverage pm.state? Is that what the firmware on the Flie is checking to change the onboard LED behavior? If not status, is there a way to learn a reliable battery state to know when to stop using my code and replace batteries?

**Comment by jsolderitsch** [comment]
As an add-on to this -- when does pm.state become 4 indicating shutdown? Is that advice to shut the CrazyFlie down -- landing as soon as possible. Does the LED color or behavior change on the front right arm for the shutdown state?

**Reply by gemenerik** [reply]
It does not appear the `shutDown` `PMState` is used. `pmSystemShutdown()` is called from the `lowPower` state after a battery critical timeout.

**Comment by gemenerik** [comment]
A solid red LED on the front right arm indicates low battery (`batteryLowVoltage`). [This](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/hal/src/pm_stm32f4.c#L425-L432) may be a good place to start investigating the related code. The decision making is thus based on voltage, not on level, although level in itself is directly based on voltage.

**Reply by jsolderitsch** [reply]
Thanks for the reply.  I will think about a more complete approach but as long as the state 3 (low voltage) is logged, then I can keep my own tick count of time in that state and then change my reported battery status from "caution" to "alarm" when my private tick count is exceeded. As long as state is reported as 0 then it would appear that the CrazyFlie is able to fly safely.
